src/utils.ts

```diff
@@ -222,3 +223,64 @@ export async function retry<T>(call: () => Promise<T>, times: number, delayS: nu
     });
   return promiseChain;
 }
+
+/**
+ * Estimates the total gas cost required to submit an unsigned (populated) transaction on-chain
+ * @param unsignedTx The unsigned transaction that this function will estimate
+ * @param relayerAddress The address that the transaction will be submitted from ()
+ * @param provider A valid ethers provider - will be used to reason the gas price
+ * @param gasPrice A manually provided gas price - if set, this function will not resolve the current gas price
+ * @returns The total gas cost to submit this transaction - i.e. gasPrice * estimatedGasUnits
+ */
+export async function estimateTotalGasRequiredByUnsignedTransaction(
+  unsignedTx: PopulatedTransaction,
+  relayerAddress: string,
+  provider: providers.Provider | L2Provider<providers.Provider>,
+  gasPrice?: BigNumberish
+): Promise<BigNumberish> {
+  const voidSigner = new VoidSigner(relayerAddress, provider);
+  // Verify if this provider has been L2Provider wrapped
+  if (isL2Provider(provider)) {
```

Comment:
nit: can we make this comment more specific that this detects whether this is an Optimism-specific provider? The way it's written now, it almost seems like all L2s get captured in this if, whereas this logic is Optimism-specific.

Alternatively, you could rename isL2Provider/L2Provider on import to something more optimism specific. Either way is cool with me!

Reply:
Updated [here](https://github.com/across-protocol/sdk-v2/pull/64/commits/d0824419500f03a0dcd73ae6c9f5dd5e96160fcf). To really highlight the difference, I added both a comment & changed the function alias